web/src/worker.ts

```diff
@@ -0,0 +1,99 @@
+import * as Comlink from "comlink";
+import init, { MapModel } from "backend";
+import type { TravelMode } from "./stores";
+import type { Position, Feature, Polygon, FeatureCollection } from "geojson";
+
+export class Backend {
+  inner: MapModel | null;
+
+  constructor() {
+    this.inner = null;
+  }
+
+  async loadFile(osmBytes: Uint8Array, progressCb: (msg: string) => void) {
+    // TODO Do we need to do this only once?
+    await init();
+
+    this.inner = new MapModel(osmBytes, progressCb);
+  }
+
+  isLoaded(): boolean {
+    return this.inner != null;
+  }
+
+  unset() {
+    this.inner = null;
+  }
+
+  getBounds(): [number, number, number, number] {
+    if (!this.inner) {
```

Comment:
Boilerplatey to have this. It's kind of up to the rest of the codebase to only call the API in the appropriate time, and it's all correct right now, but it could be possible to have a bug where a rendering layer tries to do work when a file isn't loaded